src/Storages/StorageInMemoryMetadata.cpp

```diff
@@ -493,6 +494,17 @@ namespace
 
         return res;
     }
+
+    bool isEnumSubset(const IDataType* lhs, const DataTypePtr& rhs)
+    {
+        const WhichDataType & which = WhichDataType{lhs};
+        if (!which.isEnum())
+            return false;
+        IDataTypeEnum const* enum_type = dynamic_cast<IDataTypeEnum const*>(lhs);
```

Comment:
We have typeid_cast :) Also add spaces on the both sides from asterisk)

Reply:
I can't use `typeid_cast` here because `IDataTypeEnum` is a base of actual type of `lhs` (which is `DataTypeEnum8` or `DataTypeEnum16`).

Reply:
```suggestion
        IDataTypeEnum const * enum_type = dynamic_cast<IDataTypeEnum const *>(lhs);
```